parser/src/main/java/me/vertretungsplan/parser/IphisParser.java

```diff
@@ -0,0 +1,362 @@
+        if (classIds.length > 0) {
+            if (gradesHashMap == null) {
+                throw new IOException("Change references a grade but grades are empty.");
+            }
+            final HashSet<String> classes = new HashSet<>();
+            for (String classId : classIds) {
+                classes.add(gradesHashMap.get(classId));
+            }
+            substitution.setClasses(classes);
+        }
+        // Set type
+        final String type = change.getString("aenderungsgrund").trim();
+        substitution.setType(type);
+        // Set color
+        substitution.setColor(colorProvider.getColor(type));
+        // Set covering teacher
+        final String coveringTeacherId = change.getString("id_person_verantwortlich");
+        if (!coveringTeacherId.equals("0")) {
+            if (teachersHashMap == null) {
+                throw new IOException("Change references a covering teacher but teachers are empty.");
+            }
+            substitution.setTeacher(teachersHashMap.get(coveringTeacherId));
+        }
+        // Set teacher
+        final String teacherId = change.getString("id_person_verantwortlich_orig");
+        if (!teacherId.equals("{0}")) {
+            if (teachersHashMap == null) {
+                throw new IOException("Change references a teacher but teachers are empty.");
+            }
+            if (type.equals("Vertretung") || !coveringTeacherId.equals("0")) {
+                substitution.setPreviousTeacher(teachersHashMap.get(teacherId));
+            } else {
+                substitution.setTeacher(teachersHashMap.get(teacherId));
+            }
+
+        }
+        //Set room
+        substitution.setRoom(change.optString("raum"));
+        substitution.setPreviousRoom(change.optString("raum_orig"));
+        //Set subject
+        substitution.setSubject(change.optString("fach"));
+        substitution.setPreviousSubject(change.optString("fach_orig"));
+        //Set description
+        substitution.setDesc(change.getString("information").trim());
+
+        final String startingHour = change.getString("zeit_von").replaceFirst("^0+(?!$)", "").substring(0, 5);
+        final String endingHour = change.getString("zeit_bis").replaceFirst("^0+(?!$)", "").substring(0, 5);
+        if (!startingHour.equals("") || !endingHour.equals("")) {
+            String lesson = "";
+            if (!startingHour.equals("") && endingHour.equals("")) {
+                lesson = "Ab " + startingHour;
+            }
+            if (startingHour.equals("") && !endingHour.equals("")) {
+                lesson = "Bis " + endingHour;
+            }
+            if (!startingHour.equals("") && !endingHour.equals("")) {
+                lesson = startingHour + " - " + endingHour;
+            }
+            if (!startingHour.equals(endingHour)) {
+                lesson = startingHour;
+            }
+            substitution.setLesson(lesson);
```

Comment:
zu den Lessons: wäre es denkbar, situativ entweder die Schulstunde (wie aktuell vorgesehen) bzw. die exakten Uhrzeiten bis in die App durchreichen zu können? (wie geschrieben, haben wir einige Schulen, die zeitlich sehr unterschiedliche z.B. 2. Stunden haben. Andernfalls müssten wir überlegen dort dann die entsprechende Info in den Informationstext mit aufzunehmen.

Reply:
Ja, wir können probieren, wie das in der App aussieht. Wenn es nicht passt müssten wir die Uhrzeiten in die Description einfügen.